### ISSUE TITLE
Swallows errors inside of callback

### ISSUE THREAD

**Issue by ArtskydJ** [sindresorhus/got#84]
It seems like `got` silently swallows errors.
```js
var got = require('got')

got('http://josephdykstra.com', function (err, data) {
	console.log('i see this message')
	throw new Error('lolwut')
	console.log('but not this message')
})

process.on('uncaughtException', function(err) {
	console.log('error: ' + err)
})
```
```bash
C:\Users\Michael\Github\javascript\just-login-emailer\test>node test2.js
i see this message

C:\Users\Michael\Github\javascript\just-login-emailer\test>
```

**Comment by floatdrop** [comment]
Thanks! This bug from `read-all-stream` and fixed in `3.0.1` version. Update please.

**Comment by shinnn** [comment]
:+1:
